src/components/ingredients/IngredientModal.tsx

```diff
@@ -0,0 +1,331 @@
+import Autocomplete from "./Autocomplete.tsx";
+import IngredientModalTwo from "./IngredientModalTwo.tsx";
+
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+  focusRef: RefObject<HTMLInputElement>;
+  selectedIngredients: NewIngredient[];
+  setSelectedIngredients: Dispatch<SetStateAction<NewIngredient[]>>;
+};
+
+const IngredientModal: FC<Props> = ({
+  isOpen,
+  onClose,
+  focusRef,
+  setSelectedIngredients,
+  selectedIngredients,
+}) => {
+  const { addIngredient, ingredients } = useMyStore();
+  // const purgeStorage = usePurgeStorage();
+  // const bytesInUse = useBytesInUse();
+  const [error, setError] = useState(false);
+  const [name, setName] = useState<string | null>(null);
+  const [currentIngredients, setCurrentIngredients] =
+    useState<NewIngredient[]>(ingredients);
+  const modalContainer = useRef(null);
+  const [searchQuery, setSearchQuery] = useState("");
+  const [showInput, setShowInput] = useState<boolean>(false);
+  const { data, isError } = useQuery(["data", searchQuery], () =>
+    fetchAll(searchQuery),
+  );
+  const [selectedProducts, setSelectedProducts] = useState<Product[]>([]);
+  const [step, setStep] = useState<number>(1);
+  const [selectedIngredient, setSelectedIngredient] =
+    useState<NewIngredient | null>(null);
+
+  const { productsByIds, productIds } = data ?? {
+    productIds: [],
+    productsByIds: {},
+  };
+
+  if (isError) {
+    return <div>Error</div>;
+  }
+
+  const handleIngredientClick = (ingredient: NewIngredient) => {
+    setSelectedIngredients([...selectedIngredients, ingredient]);
+    // const updatedIngredients = ingredients.filter(
+    //   (item) => !selectedIngredients.includes(item) && item !== ingredient,
+    // );
+    // setCurrentIngredients(updatedIngredients);
+  };
+
+  const handleSave = () => {
+    if (name != null && selectedProducts.length > 0) {
+      const updatedSelectedProducts: SimpleIngredient[] = selectedProducts.map(
+        (product) => {
+          const { id, preferred } = product;
+          return { id, preferred };
+        },
+      );
+      const newIngredient: NewIngredient = {
+        name: name,
+        selectedProducts: updatedSelectedProducts,
+        id: Date.now().toString(36),
+      };
+      addIngredient(newIngredient);
+      handleIngredientClick(newIngredient);
+      setShowInput(false);
+      setSelectedProducts([]);
+      setSearchQuery("");
+      setName(null);
+      setStep(1);
+      onClose();
+    } else if (name != null && selectedProducts.length === 0) {
+      setError(true);
+    }
+  };
+
+  useEffect(() => {
+    const filteredIngredients = ingredients.filter(
+      (ingredient) =>
+        !selectedIngredients.some(
+          (selected) => selected.id === ingredient.id,
+        ) &&
+        (!selectedIngredient || selectedIngredient.id !== ingredient.id),
+    );
+
+    setCurrentIngredients(filteredIngredients);
+  }, [selectedIngredients, selectedIngredient]);
```

Comment:
Jak pracujeme s `currentIngredients` musim jeste nastudovat, na prvni dobrou se mi tohle moc nepozdava. Prijde mi, ze takhle "reaktivni" by to byt nemelo.. cekal bych, ze tyhle veci se vzdy budou resit v nejake "user akci" v onClicku, etc..